sql/firefox_accounts_exact_mau28_raw_v1.sql

```diff
@@ -0,0 +1,38 @@
+WITH
+  inactive_days AS (
+    SELECT
+      *,
+      DATE_DIFF(submission_date, date_last_seen, DAY) AS _inactive_days,
+      DATE_DIFF(submission_date, date_last_seen_in_tier1_country, DAY) AS _inactive_days_tier1
+    FROM
+      fxa_users_last_seen_v1
+  )
+
+SELECT
+  submission_date,
+  CURRENT_DATETIME() AS generated_time,
+  COUNTIF(_inactive_days < 28) AS mau,
+  COUNTIF(_inactive_days < 7) AS wau,
+  COUNTIF(_inactive_days < 1) AS dau,
+  -- We are generally using an "exclusive dimensions" methodology where only
+  -- the last country observed for a user is considered for determining whether
+  -- they contribute to tier 1 MAU, but we also include an "inclusive" tier 1
+  -- mau calculation that we have previously been using for KPI calculations on
+  -- FxA data; a user is included in this calculation if it appeared in a tier 1
+  -- country during any of the 28 days in the MAU window.
+  COUNTIF(_inactive_days_tier1 < 28) AS mau_tier1_inclusive,
+  -- We hash user_ids into 20 buckets to aid in computing
+  -- confidence intervals for mau/wau/dau sums; the particular hash
+  -- function and number of buckets is subject to change in the future.
+  MOD(ABS(FARM_FINGERPRINT(user_id)), 20) AS id_bucket,
```

Comment:
drive-by: should we shift to non-negative range rather than taking `ABS(...)` here?

Reply:
I think it would be complex to do so. `FARM_FINGERPRINT` returns a signed INT64 and BQ does not have any larger or unsigned integer variants, so we don't have a type to shift into.

Is your concern that ABS is introducing a bias between buckets?

Reply:
Yep, that's my concern. It may not matter, and I agree that working around it seems overly complex here.

Reply:
this would shift mod to positive, and guarantee we don't get a int overflow if we hit `-0x8000000000000000`:
```suggestion
  MOD(MOD(FARM_FINGERPRINT(user_id), 20) + 20, 20) AS id_bucket,
```

Reply:
I'm going to stick with the simpler form for now. I've broken out the hash function discussion to https://github.com/mozilla/bigquery-etl/issues/36 so it can be evaluated as a separate project in the future.